modules/tinymce/src/core/main/ts/selection/ElementSelection.ts

```diff
@@ -89,58 +87,60 @@ const getNode = (root: Element, rng: Range): Element => {
     }
   }
 
-  if (elm && elm.nodeType === 3) {
+  if (NodeType.isText(elm)) {
     return elm.parentNode;
   }
 
   return elm;
 };
 
 const getSelectedBlocks = (dom: DOMUtils, rng: Range, startElm?: Element, endElm?: Element): Element[] => {
-  let node;
-  const selectedBlocks = [];
+  const selectedBlocks: Element[] = [];
 
   const root = dom.getRoot();
-  startElm = dom.getParent(startElm || getStart(root, rng, rng.collapsed), dom.isBlock);
-  endElm = dom.getParent(endElm || getEnd(root, rng, rng.collapsed), dom.isBlock);
+  const start = dom.getParent(startElm || getStart(root, rng, rng.collapsed), dom.isBlock);
+  const end = dom.getParent(endElm || getEnd(root, rng, rng.collapsed), dom.isBlock);
 
-  if (startElm && startElm !== root) {
-    selectedBlocks.push(startElm);
+  if (start && start !== root) {
+    selectedBlocks.push(start);
   }
 
-  if (startElm && endElm && startElm !== endElm) {
-    node = startElm;
+  if (start && end && start !== end) {
+    let node: Node | undefined = start;
 
-    const walker = new DomTreeWalker(startElm, root);
-    while ((node = walker.next()) && node !== endElm) {
+    const walker = new DomTreeWalker(start, root);
+    while ((node = walker.next()) && node !== end) {
       if (dom.isBlock(node)) {
         selectedBlocks.push(node);
       }
     }
   }
 
-  if (endElm && startElm !== endElm && endElm !== root) {
-    selectedBlocks.push(endElm);
+  if (end && start !== end && end !== root) {
+    selectedBlocks.push(end);
   }
 
   return selectedBlocks;
 };
 
-const select = (dom, node: Node, content?: boolean) => Optional.from(node).map((node) => {
-  const idx = dom.nodeIndex(node);
-  const rng = dom.createRng();
+const select = (dom: DOMUtils, node: Node | null, content?: boolean) =>
+  Optional.from(node).bind((node) =>
+    Optional.from(node.parentNode).map((parent) => {
```

Comment:
Minor code change here to account for `node.parentNode` being nullable.

Reply:
Could this be collapsed to `Optional.from(node?.parentNode)` and avoid the extra level of nesting?

Reply:
ugh of course it does 😩